relayer/checkpoint/checkpoint.go

```diff
@@ -43,30 +47,27 @@ func NewCheckpointManager(
 		zap.Uint64("startingHeight", startingHeight),
 	)
 
-	committedHeight := startingHeight
 	storedHeight, err := database.GetLatestProcessedBlockHeight(db, relayerID)
 	if err != nil && !database.IsKeyNotFoundError(err) {
 		logger.Error(
 			"Failed to get latest processed block height",
 			zap.Error(err),
 			zap.String("relayerID", relayerID.ID.String()),
 		)
-		return nil
+		return nil, fmt.Errorf("failed to get the latest processed block height: %v", err)
 	}
 
-	if storedHeight > committedHeight {
-		committedHeight = storedHeight
-	}
+	committedHeight := math.Max(float64(storedHeight), float64(startingHeight))
```

Comment:
Whoops, sorry I forgot `math.Max()` only took floats... I don't think we want to cast here. We should just write a helper function for uint64s (or a generic function if you're feeling ambitious).